app/javascript/src/components/Player.tsx

```diff
@@ -202,6 +209,29 @@ class Player extends React.Component<PlayerProps, PlayerState> {
     )
   }
 
+  private goBack() {
+    return (
+      <a className="SlantButton" id="back-button" onClick={this.setPreviousState.bind(this)} >
+        Back
+      </a>
+    )
+  }
+
+  private setPreviousState() {
+    let previous = this.state.previousState
+    previous.splice(-1)
+    let last = this.state.previousState.length - 1
```

Comment:
**functions**:

- `goBack` -> `goBackButton`
  - "goBack" sounds like it's describing code that would actually do the going back. Instead that function is just rendering a button
- `setPreviousState` -> `goBack` or `revertToPreviousState`
  - "set previous state" sounds to me like you're just updating one piece of the state, but you're doing much more than that

** state variable**:
- `previousState` -> `previousStates` or `history`
  - "previous state" sounds like it's a singular object. Since that state variable is an array of previous states, either pluralize the name, or use a name that implies there's a string of them

**variables in this function**:
- `previous` -> `newPreviousStates` or `newHistory`
  - "new" meaning you're about the change the component state, followed by the name of whatever the state variable is to be consistent
- `last` -> either `lastIndex`, OR better yet, grab the most recent state off the stack and store that instead of passing storing the list index variable
  - `previousState = this.state.previousState[this.state.previousState.length - 1]`
  - OR if you clone the state variable, you could just `pop` off the last one in the clone:
  ```js
  let newPreviousStates = this.state.previousStates
  let lastPreviousState = newPreviousStates.pop()

  this.setState({
    focus: lastPreviousState.focus,
    currentItems: lastPreviousState.currentItems,
    currentStats: lastPreviousState.currentStats,
    previousStates: newPreviousStates
  })
  ```

  or maybe even get fancy with the ending there:
  ```js
  this.setState({
    ...lastPreviousState,
    previousStates: newPreviousStates
  })
  ```